### PR TITLE
smb: optimize smb mount performance by avoiding stat checks during initialization

### PR DESCRIPTION
**Pull request by sudiptob2**
#### What is the purpose of this change?
Detect whether a mountpoint is a directory based on the presence of a trailing slash. Improves performance during the initial mount for SMB-backed by avoiding unnecessary stat checks.

#### Was the change discussed in an issue or in the forum before?
This issue is discussed in detail in this forum post: https://forum.rclone.org/t/vfs-mount-to-smb-gets-rate-limited-due-to-initial-stat-checks/52479

#### Checklist

- [ ] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [ ] I have added tests for all changes in this PR if appropriate.
- [ ] I have added documentation for the changes if appropriate.
- [ ] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [ ] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

File: backend/smb/smb.go [hunk 1]
```diff
@@ -894,6 +902,11 @@ func ensureSuffix(s, suffix string) string {
 	return s + suffix
 }
 
+// IsPathDir determines if a path represents a directory based on trailing slash
+func IsPathDir(path string) bool {
```
Comment (darthShadow) [inline backend/smb/smb.go]: nit: `isPathDir` for a private function.

File: backend/smb/smb_test.go [hunk 2]
```diff
@@ -51,3 +51,40 @@ func TestIntegration3(t *testing.T) {
...
+
+		// Edge cases
+		{"share//", true},
+		{"share///", true},
+		{"share/dir//", true},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.path, func(t *testing.T) {
+			result := smb.IsPathDir(tt.path)
```
Comment (darthShadow) [inline backend/smb/smb_test.go]: I think you should be able to use `isPathDir` directly here since they are in the same folder if you rename the package to just `smb` rather than `smb_test`.

Comment (sudiptob2) [reply backend/smb/smb_test.go]: Hey, thanks! Looking at it now, I think the `smb_test.go` is supposed to provide black box integration tests. So I moved the tests for `IsPathDir` into a different file.

**Review by ncw — COMMENTED** [review]
That looks great thank you :-)

Can you rename the file `smb_unit_test.go` to `smb_internal_test.go` then it will fit in with the naming scheme in the other backends.

Once you've done that I'll merge!

**Comment by sudiptob2** [comment]
@ncw Thank you so much, renamed the test file.
